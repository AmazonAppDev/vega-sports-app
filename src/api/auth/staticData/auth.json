{
  "auth": {
    "id": "nmbsd-234sdc-678-jklsdf",
    "name": "Freida",
    "surname": "Gomam",
    "email": "fgomam@fakemail.com",
    "address": "Nowhere 5 USA",
    "image": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/5p.jpg",
    "profiles": [
      {
        "id": "bada55-1dl00k-4tm33",
        "displayName": "Freida",
        "image": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/6p.jpg"
      },
      {
        "id": "l00k-1nt0-s0uL",
        "displayName": "Jennifer",
        "image": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/7p.jpg"
      }
    ]
  }
}
