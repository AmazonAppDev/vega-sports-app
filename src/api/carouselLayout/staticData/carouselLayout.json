{
  "carousellayout": [
    {
      "id": "e77effd2-60fd-4a75-9921-ee1596c4d248",
      "carouselType": "hero",
      "carouselTitle": null,
      "endpoint": "suggestedforyou"
    },
    {
      "id": "fac0db1d-60ee-4ff9-9c3f-fc1ca24fa6be",
      "carouselType": "square",
      "carouselTitle": "Live TV: fresh news from events",
      "endpoint": "livestreams"
    },
    {
      "id": "cf5bafe0-023b-4ff9-9c3f-fc1ca24fa6be",
      "carouselType": "card",
      "carouselTitle": "Sport Teams Hall of Fame",
      "endpoint": "teams"
    },
    {
      "id": "068f3841-023b-4de8-85d1-f410d1407528",
      "carouselType": "square",
      "carouselTitle": "Documentaries on Sport Stars",
      "endpoint": "documentaries"
    },
    {
      "id": "9fc13a87-af56-42ef-a66e-b882a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c91f1389-845c-476d-820b-1aa388e7bf1a4",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    },
    {
      "id": "5d21039f-3ba5-491b-b481-324ceb742ac71",
      "carouselType": "card",
      "carouselTitle": "Champions League",
      "endpoint": "documentaries"
    },
    {
      "id": "fac0db1d-60ee-4ff9-9c3f-fc15ca24fa6be",
      "carouselType": "square",
      "carouselTitle": "Live TV: fresh news from events",
      "endpoint": "livestreams"
    },
    {
      "id": "cf5bafe0-023b-4ff9-9c3f-6fc1ca24fa6be",
      "carouselType": "card",
      "carouselTitle": "Sport Teams Hall of Fame",
      "endpoint": "teams"
    },
    {
      "id": "06844f3841-023b-4de8-85d1-7f410d14aa07528",
      "carouselType": "square",
      "carouselTitle": "Documentaries on Sport Stars",
      "endpoint": "documentaries"
    },
    {
      "id": "9fc1113a87-af56-42ef-a66e8-b8jj8a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c91f11389-845c-476d-8290b-1aajj88e7bf1a4",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    },
    {
      "id": "5d221039f-3ba5-491b-b4081-32cejjb742ac71",
      "carouselType": "card",
      "carouselTitle": "Champions League",
      "endpoint": "documentaries"
    },
    {
      "id": "9fc132a87-af56-42ef-a66e-bgh882a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c9177f14389-845c-476d-820b-s",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    },
    {
      "id": "9fc166113a87-af56-42ef-a66e8-b8jj8a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c91f5511389-845c-476d-8290b-1aajj88e7bf1a4",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    },
    {
      "id": "5d22441039f-3ba5-491b-b4081-32cejjb742ac71",
      "carouselType": "card",
      "carouselTitle": "Champions League",
      "endpoint": "documentaries"
    },
    {
      "id": "9fc13233a87-af56-42ef-a66e-bgh882a88540cc9",
      "carouselType": "card",
      "carouselTitle": "Roland Garros",
      "endpoint": "teams"
    },
    {
      "id": "c91f1422389-845c-476d-820b-s",
      "carouselType": "square",
      "carouselTitle": "Interviews with Idols",
      "endpoint": "livestreams"
    }
  ]
}
