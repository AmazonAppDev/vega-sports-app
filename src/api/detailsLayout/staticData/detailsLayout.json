{
  "detailslayout": [
    {
      "id": "documentaries",
      "layout_type": "documentaries",
      "layout_title": "Sports Documentaries",
      "layoutElements": [
        {
          "layoutElements": [
            {
              "layoutElements": [
                {
                  "id": "containera-imagetile1",
                  "description": "Explore the journey of legendary athletes.",
                  "image": {
                    "url": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/30.jpg"
                  },
                  "displayProps": {
                    "size": "custom",
                    "customWidth": 500,
                    "customHeight": 200
                  },
                  "targetUrl": {
                    "url": "details/documentaries"
                  },
                  "title": "Athlete Stories",
                  "elementType": "DCImageTile"
                },
                {
                  "id": "containera-imagetile2",
                  "description": "Read about the top stars.",
                  "image": {
                    "url": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/31.jpg"
                  },
                  "displayProps": {
                    "size": "custom",
                    "customWidth": 500,
                    "customHeight": 200
                  },
                  "targetUrl": {
                    "url": "details/documentaries"
                  },
                  "title": "Top stars",
                  "elementType": "DCImageTile"
                }
              ],
              "displayProps": {
                "horizontalSpacing": 0,
                "flexDirection": "row"
              },
              "title": null,
              "elementType": "DCContentContainer"
            },
            {
              "layoutElements": [
                {
                  "description": "A closer look at iconic sports moments.",
                  "displayProps": {
                    "size": "small"
                  },
                  "id": "containera-subcontainer-imagetile1",
                  "image": {
                    "url": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/32.jpg"
                  },
                  "targetUrl": {
                    "url": "details/documentaries"
                  },
                  "title": "Iconic Moments",
                  "elementType": "DCImageTile"
                },
                {
                  "description": "Behind the scenes of championship games.",
                  "displayProps": {
                    "size": "small"
                  },
                  "id": "containera-subcontainer-imagetile2",
                  "image": {
                    "url": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/33.jpg"
                  },
                  "targetUrl": {
                    "url": "details/documentaries"
                  },
                  "title": "Championship Stories",
                  "elementType": "DCImageTile"
                },
                {
                  "description": "The untold stories of rising stars.",
                  "displayProps": {
                    "size": "small"
                  },
                  "id": "containera-subcontainer-imagetile3",
                  "image": {
                    "url": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/35.jpg"
                  },
                  "targetUrl": {
                    "url": "details/documentaries"
                  },
                  "title": "Rising Stars",
                  "elementType": "DCImageTile"
                },
                {
                  "description": "Team insights and player stats.",
                  "displayProps": {
                    "size": "small"
                  },
                  "id": "sectionb-tilewithimage2",
                  "image": {
                    "url": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/36.jpg"
                  },
                  "targetUrl": {
                    "url": "details/teams"
                  },
                  "title": "Team Profiles",
                  "elementType": "DCImageTile"
                }
              ],
              "displayProps": {
                "horizontalSpacing": 18,
                "flexDirection": "row",
                "verticalSpacing": 20
              },
              "title": null,
              "elementType": "DCContentContainer"
            },
            {
              "elementType": "DCText",
              "text": "Athlete stories provide an insight into the lives of sports stars beyond the game. They highlight the challenges, sacrifices, and triumphs that shape their journeys to success. From overcoming injuries to breaking records, these stories inspire fans and aspiring athletes alike. They often reveal the personal struggles and resilience that drive their passion for the sport. Athlete stories also celebrate their achievements, both on and off the field, showcasing their impact on the community. Through these narratives, we connect with the human side of sports, finding motivation and admiration in their experiences.",
              "displayProps": {
                "variant": "body",
                "alignContent": "center",
                "alignItems": "center",
                "justifyContent": "center"
              },
              "id": "container-body1"
            }
          ],
          "displayProps": {
            "horizontalSpacing": 18
          },
          "title": "Documentary Highlights",
          "elementType": "DCContentContainer"
        }
      ]
    },
    {
      "id": "livestreams",
      "layout_type": "livestreams",
      "layout_title": "Live Matches",
      "layoutElements": [
        {
          "layoutElements": [
            {
              "description": "Team 1 is ready to dominate the field.",
              "displayProps": {
                "size": "medium"
              },
              "id": "containerlivestreams-imagetile1",
              "image": {
                "url": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/38.jpg",
                "targetUrl": "team1.thumbnail"
              },
              "targetUrl": {
                "url": "details/teams"
              },
              "title": "Team 1",
              "titleTarget": "team1.name",
              "elementType": "DCImageTile"
            },
            {
              "displayProps": {
                "variant": "headline",
                "alignContent": "center",
                "alignItems": "center",
                "justifyContent": "center"
              },
              "id": "containerlivestreams-text1",
              "text": "VS",
              "elementType": "DCText"
            },
            {
              "description": "Team 2 is bringing their A-game.",
              "id": "containerlivestreams-imagetile2",
              "displayProps": {
                "size": "medium"
              },
              "image": {
                "url": "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/250x250/39.jpg",
                "targetUrl": "team2.thumbnail"
              },
              "targetUrl": {
                "url": "details/teams"
              },
              "title": "Team 2",
              "titleTarget": "team2.name",
              "elementType": "DCImageTile"
            }
          ],
          "displayProps": {
            "horizontalSpacing": 18,
            "flexDirection": "row"
          },
          "title": null,
          "elementType": "DCContentContainer"
        },
        {
          "layoutElements": [
            {
              "displayProps": {
                "variant": "headline"
              },
              "id": "containerlivestreams-text1",
              "text": "Upcoming Match Highlights",
              "textTarget": "headline",
              "elementType": "DCText"
            },
            {
              "displayProps": {
                "variant": "body"
              },
              "id": "containerlivestreams-text2",
              "text": "Get ready for an exciting clash between two top teams. Stay tuned for live updates and in-depth analysis.",
              "textTarget": "description",
              "elementType": "DCText"
            }
          ],
          "displayProps": {
            "horizontalSpacing": 50,
            "verticalSpacing": 30
          },
          "title": null,
          "elementType": "DCContentContainer"
        }
      ]
    },
    {
      "id": "suggestedforyou",
      "layout_type": "suggestedforyou",
      "layout_title": "Recommended for You",
      "layoutElements": [
        {
          "layoutElements": [
            {
              "displayProps": {
                "variant": "headline"
              },
              "id": "containerlivestreams-text1",
              "text": "Curated Sports Content Just for You",
              "elementType": "DCText"
            }
          ],
          "title": null,
          "elementType": "DCContentContainer"
        }
      ]
    },
    {
      "id": "teams",
      "layout_type": "teams",
      "layout_title": "Team Profiles",
      "layoutElements": [
        {
          "id": "containerteams2",
          "layoutElements": [
            {
              "id": "containerteams2-1",
              "layoutElements": [
                {
                  "description": "",
                  "displayProps": {
                    "size": "custom",
                    "customWidth": 300,
                    "customHeight": 450
                  },
                  "id": "containerteams-imagetile1",
                  "image": {
                    "targetUrl": "thumbnail"
                  },
                  "titleTarget": "teamName",
                  "targetUrl": {
                    "url": "details/teams"
                  },
                  "elementType": "DCImageTile"
                }
              ],
              "displayProps": {
                "horizontalSpacing": 18,
                "flex": 0
              },
              "title": null,
              "elementType": "DCContentContainer"
            },
            {
              "id": "containerteams2-2",
              "layoutElements": [
                {
                  "displayProps": {
                    "variant": "body"
                  },
                  "id": "containerteams2-text1",
                  "text": "Sport type:",
                  "title": "DCText 1",
                  "elementType": "DCText"
                },
                {
                  "displayProps": {
                    "variant": "title",
                    "horizontalSpacing": 24
                  },
                  "id": "containerteams2-text2",
                  "text": "N/A",
                  "textTarget": "sportType",
                  "title": "DCText 2",
                  "elementType": "DCText"
                },
                {
                  "displayProps": {
                    "variant": "body",
                    "topSpacing": 24
                  },
                  "id": "containerteams2-text3",
                  "text": "Coach:",
                  "title": "DCText 3",
                  "elementType": "DCText"
                },
                {
                  "displayProps": {
                    "variant": "title",
                    "horizontalSpacing": 24
                  },
                  "id": "containerteams2-text4",
                  "text": "N/A",
                  "textTarget": "coachName",
                  "title": "DCText 4",
                  "elementType": "DCText"
                },
                {
                  "displayProps": {
                    "variant": "body",
                    "topSpacing": 24
                  },
                  "id": "containerteams2-text5",
                  "text": "Home stadium:",
                  "title": "DCText 5",
                  "elementType": "DCText"
                },
                {
                  "displayProps": {
                    "variant": "title",
                    "horizontalSpacing": 24
                  },
                  "id": "containerteams2-text6",
                  "text": "N/A",
                  "textTarget": "homeStadium",
                  "title": "DCText 6",
                  "elementType": "DCText"
                },
                {
                  "displayProps": {
                    "variant": "body",
                    "topSpacing": 24
                  },
                  "id": "containerteams2-text7",
                  "text": "Team color:",
                  "title": "DCText 7",
                  "elementType": "DCText"
                },
                {
                  "displayProps": {
                    "variant": "title",
                    "horizontalSpacing": 24
                  },
                  "id": "containerteams2-text8",
                  "text": "N/A",
                  "textTarget": "teamColor",
                  "title": "DCText 8",
                  "elementType": "DCText"
                }
              ],
              "displayProps": {
                "alignSelf": "flex-start",
                "justifyContent": "flex-start",
                "alignContent": "flex-start"
              },
              "title": null,
              "elementType": "DCContentContainer"
            }
          ],
          "displayProps": {
            "horizontalSpacing": 18,
            "verticalSpacing": 36,
            "flexDirection": "row",
            "alignSelf": "flex-start",
            "justifyContent": "flex-start",
            "alignContent": "flex-start"
          },
          "title": null,
          "elementType": "DCContentContainer"
        }
      ]
    }
  ]
}
