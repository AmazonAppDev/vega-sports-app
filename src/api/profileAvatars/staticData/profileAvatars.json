{
  "profileavatars": [
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/1p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/2p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/3p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/4p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/5p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/6p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/7p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/8p.jpg",
    "https://d1v0fxmwkpxbrg.cloudfront.net/sports-app/images/profiles/9p.jpg"
  ]
}
