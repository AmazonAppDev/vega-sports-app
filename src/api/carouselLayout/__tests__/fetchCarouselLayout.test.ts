import { useQuery } from '@tanstack/react-query';
import { renderHook } from '@testing-library/react-native';

import { AppConfig } from '@AppServices/appConfig';
import {
  useCarouselLayout,
  fetchCarouselLayoutApiCall,
} from '../fetchCarouselLayout';
import staticData from '../staticData/carouselLayout.json';

jest.mock('@tanstack/react-query');

const mockRQ = useQuery as jest.Mock;

describe('fetchDocumentaries', () => {
  beforeAll(() => {
    require('jest-fetch-mock').enableMocks();
  });

  afterEach(() => {
    fetchMock.resetMocks();
  });

  describe('fetchCarouselLayoutApiCall', () => {
    it('should return parsed documentaries data on success', async () => {
      fetchMock.mockResponseOnce(
        JSON.stringify({ data: staticData.carousellayout }),
        {
          status: 200,
        },
      );

      const result = await fetchCarouselLayoutApiCall();

      expect(result).toMatchSnapshot();
    });

    it('should throw error for 400 status', async () => {
      jest.spyOn(AppConfig, 'isUsingStaticData').mockReturnValueOnce(false);

      fetchMock.mockResponseOnce(JSON.stringify({}), { status: 400 });

      await expect(fetchCarouselLayoutApiCall()).rejects.toThrow(
        `fetchCarouselLayoutApiCall(): resources does not exists for endpoint 'carousellayout'`,
      );
    });

    it('should throw error for unexpected Api status', async () => {
      jest.spyOn(AppConfig, 'isUsingStaticData').mockReturnValueOnce(false);

      // @ts-expect-error intentionally break TS
      fetchMock.mockResponseOnce(null);

      await expect(fetchCarouselLayoutApiCall()).rejects.toThrow(
        `Perform a request failed`,
      );
    });

    it('should throw error for other non-200 status', async () => {
      jest.spyOn(AppConfig, 'isUsingStaticData').mockReturnValueOnce(false);

      fetchMock.mockResponseOnce(JSON.stringify({}), { status: 500 });

      await expect(fetchCarouselLayoutApiCall()).rejects.toThrow(
        `fetchCarouselLayoutApiCall(): failed to fetch data from endpoint 'carousellayout'`,
      );
    });
  });

  describe('useCarouselLayout', () => {
    it('should return carousel layout data and no error on successful fetch', async () => {
      fetchMock.mockResponseOnce(
        JSON.stringify({ data: staticData.carousellayout }),
        {
          status: 200,
        },
      );

      mockRQ.mockReturnValueOnce({
        data: await fetchCarouselLayoutApiCall(),
        error: null,
        isLoading: false,
      });

      const { result } = renderHook(() => useCarouselLayout());

      expect(result.current.data).toMatchSnapshot();
      expect(result.current.error).toBe(null);
      expect(result.current.isLoading).toBe(false);
    });

    it('should return isLoading as true before data loads', () => {
      mockRQ.mockReturnValueOnce({
        data: null,
        error: null,
        isLoading: true,
      });

      const { result } = renderHook(() => useCarouselLayout());

      expect(result.current.isLoading).toBe(true);
      expect(result.current.data).toBe(null);
      expect(result.current.error).toBe(null);
    });

    it('should return error when fetch fails', () => {
      const mockError = new Error('Network error');

      mockRQ.mockReturnValueOnce({
        data: null,
        error: mockError,
        isLoading: false,
      });

      const { result } = renderHook(() => useCarouselLayout());

      expect(result.current.error).toBe(mockError);
      expect(result.current.data).toBe(null);
      expect(result.current.isLoading).toBe(false);
    });
  });
});
