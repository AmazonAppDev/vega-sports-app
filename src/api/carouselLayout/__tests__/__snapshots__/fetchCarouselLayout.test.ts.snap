// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchDocumentaries fetchCarouselLayoutApiCall should return parsed documentaries data on success 1`] = `
[
  {
    "carouselTitle": null,
    "carouselType": "hero",
    "endpoint": "suggestedforyou",
    "itemId": "e77effd2-60fd-4a75-9921-ee1596c4d248",
  },
  {
    "carouselTitle": "Live TV: fresh news from events",
    "carouselType": "square",
    "endpoint": "livestreams",
    "itemId": "fac0db1d-60ee-4ff9-9c3f-fc1ca24fa6be",
  },
  {
    "carouselTitle": "Sport Teams Hall of Fame",
    "carouselType": "card",
    "endpoint": "teams",
    "itemId": "cf5bafe0-023b-4ff9-9c3f-fc1ca24fa6be",
  },
  {
    "carouselTitle": "Documentaries on Sport Stars",
    "carouselType": "square",
    "endpoint": "documentaries",
    "itemId": "068f3841-023b-4de8-85d1-f410d1407528",
  },
  {
    "carouselTitle": "Roland Garros",
    "carouselType": "card",
    "endpoint": "teams",
    "itemId": "9fc13a87-af56-42ef-a66e-b882a88540cc9",
  },
  {
    "carouselTitle": "Interviews with Idols",
    "carouselType": "square",
    "endpoint": "livestreams",
    "itemId": "c91f1389-845c-476d-820b-1aa388e7bf1a4",
  },
  {
    "carouselTitle": "Champions League",
    "carouselType": "card",
    "endpoint": "documentaries",
    "itemId": "5d21039f-3ba5-491b-b481-324ceb742ac71",
  },
  {
    "carouselTitle": "Live TV: fresh news from events",
    "carouselType": "square",
    "endpoint": "livestreams",
    "itemId": "fac0db1d-60ee-4ff9-9c3f-fc15ca24fa6be",
  },
  {
    "carouselTitle": "Sport Teams Hall of Fame",
    "carouselType": "card",
    "endpoint": "teams",
    "itemId": "cf5bafe0-023b-4ff9-9c3f-6fc1ca24fa6be",
  },
  {
    "carouselTitle": "Documentaries on Sport Stars",
    "carouselType": "square",
    "endpoint": "documentaries",
    "itemId": "06844f3841-023b-4de8-85d1-7f410d14aa07528",
  },
  {
    "carouselTitle": "Roland Garros",
    "carouselType": "card",
    "endpoint": "teams",
    "itemId": "9fc1113a87-af56-42ef-a66e8-b8jj8a88540cc9",
  },
  {
    "carouselTitle": "Interviews with Idols",
    "carouselType": "square",
    "endpoint": "livestreams",
    "itemId": "c91f11389-845c-476d-8290b-1aajj88e7bf1a4",
  },
  {
    "carouselTitle": "Champions League",
    "carouselType": "card",
    "endpoint": "documentaries",
    "itemId": "5d221039f-3ba5-491b-b4081-32cejjb742ac71",
  },
  {
    "carouselTitle": "Roland Garros",
    "carouselType": "card",
    "endpoint": "teams",
    "itemId": "9fc132a87-af56-42ef-a66e-bgh882a88540cc9",
  },
  {
    "carouselTitle": "Interviews with Idols",
    "carouselType": "square",
    "endpoint": "livestreams",
    "itemId": "c9177f14389-845c-476d-820b-s",
  },
  {
    "carouselTitle": "Roland Garros",
    "carouselType": "card",
    "endpoint": "teams",
    "itemId": "9fc166113a87-af56-42ef-a66e8-b8jj8a88540cc9",
  },
  {
    "carouselTitle": "Interviews with Idols",
    "carouselType": "square",
    "endpoint": "livestreams",
    "itemId": "c91f5511389-845c-476d-8290b-1aajj88e7bf1a4",
  },
  {
    "carouselTitle": "Champions League",
    "carouselType": "card",
    "endpoint": "documentaries",
    "itemId": "5d22441039f-3ba5-491b-b4081-32cejjb742ac71",
  },
  {
    "carouselTitle": "Roland Garros",
    "carouselType": "card",
    "endpoint": "teams",
    "itemId": "9fc13233a87-af56-42ef-a66e-bgh882a88540cc9",
  },
  {
    "carouselTitle": "Interviews with Idols",
    "carouselType": "square",
    "endpoint": "livestreams",
    "itemId": "c91f1422389-845c-476d-820b-s",
  },
]
`;

exports[`fetchDocumentaries useCarouselLayout should return carousel layout data and no error on successful fetch 1`] = `
[
  {
    "carouselTitle": null,
    "carouselType": "hero",
    "endpoint": "suggestedforyou",
    "itemId": "e77effd2-60fd-4a75-9921-ee1596c4d248",
  },
  {
    "carouselTitle": "Live TV: fresh news from events",
    "carouselType": "square",
    "endpoint": "livestreams",
    "itemId": "fac0db1d-60ee-4ff9-9c3f-fc1ca24fa6be",
  },
  {
    "carouselTitle": "Sport Teams Hall of Fame",
    "carouselType": "card",
    "endpoint": "teams",
    "itemId": "cf5bafe0-023b-4ff9-9c3f-fc1ca24fa6be",
  },
  {
    "carouselTitle": "Documentaries on Sport Stars",
    "carouselType": "square",
    "endpoint": "documentaries",
    "itemId": "068f3841-023b-4de8-85d1-f410d1407528",
  },
  {
    "carouselTitle": "Roland Garros",
    "carouselType": "card",
    "endpoint": "teams",
    "itemId": "9fc13a87-af56-42ef-a66e-b882a88540cc9",
  },
  {
    "carouselTitle": "Interviews with Idols",
    "carouselType": "square",
    "endpoint": "livestreams",
    "itemId": "c91f1389-845c-476d-820b-1aa388e7bf1a4",
  },
  {
    "carouselTitle": "Champions League",
    "carouselType": "card",
    "endpoint": "documentaries",
    "itemId": "5d21039f-3ba5-491b-b481-324ceb742ac71",
  },
  {
    "carouselTitle": "Live TV: fresh news from events",
    "carouselType": "square",
    "endpoint": "livestreams",
    "itemId": "fac0db1d-60ee-4ff9-9c3f-fc15ca24fa6be",
  },
  {
    "carouselTitle": "Sport Teams Hall of Fame",
    "carouselType": "card",
    "endpoint": "teams",
    "itemId": "cf5bafe0-023b-4ff9-9c3f-6fc1ca24fa6be",
  },
  {
    "carouselTitle": "Documentaries on Sport Stars",
    "carouselType": "square",
    "endpoint": "documentaries",
    "itemId": "06844f3841-023b-4de8-85d1-7f410d14aa07528",
  },
  {
    "carouselTitle": "Roland Garros",
    "carouselType": "card",
    "endpoint": "teams",
    "itemId": "9fc1113a87-af56-42ef-a66e8-b8jj8a88540cc9",
  },
  {
    "carouselTitle": "Interviews with Idols",
    "carouselType": "square",
    "endpoint": "livestreams",
    "itemId": "c91f11389-845c-476d-8290b-1aajj88e7bf1a4",
  },
  {
    "carouselTitle": "Champions League",
    "carouselType": "card",
    "endpoint": "documentaries",
    "itemId": "5d221039f-3ba5-491b-b4081-32cejjb742ac71",
  },
  {
    "carouselTitle": "Roland Garros",
    "carouselType": "card",
    "endpoint": "teams",
    "itemId": "9fc132a87-af56-42ef-a66e-bgh882a88540cc9",
  },
  {
    "carouselTitle": "Interviews with Idols",
    "carouselType": "square",
    "endpoint": "livestreams",
    "itemId": "c9177f14389-845c-476d-820b-s",
  },
  {
    "carouselTitle": "Roland Garros",
    "carouselType": "card",
    "endpoint": "teams",
    "itemId": "9fc166113a87-af56-42ef-a66e8-b8jj8a88540cc9",
  },
  {
    "carouselTitle": "Interviews with Idols",
    "carouselType": "square",
    "endpoint": "livestreams",
    "itemId": "c91f5511389-845c-476d-8290b-1aajj88e7bf1a4",
  },
  {
    "carouselTitle": "Champions League",
    "carouselType": "card",
    "endpoint": "documentaries",
    "itemId": "5d22441039f-3ba5-491b-b4081-32cejjb742ac71",
  },
  {
    "carouselTitle": "Roland Garros",
    "carouselType": "card",
    "endpoint": "teams",
    "itemId": "9fc13233a87-af56-42ef-a66e-bgh882a88540cc9",
  },
  {
    "carouselTitle": "Interviews with Idols",
    "carouselType": "square",
    "endpoint": "livestreams",
    "itemId": "c91f1422389-845c-476d-820b-s",
  },
]
`;
